internal/local_utils/local_utils.go

```diff
@@ -0,0 +1,149 @@
+//go:build remote && (amd64 || arm64)
+
+package local_utils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/url"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/containers/podman/v5/pkg/bindings"
+	"github.com/containers/podman/v5/pkg/machine/define"
+	"github.com/containers/podman/v5/pkg/machine/env"
+	"github.com/containers/podman/v5/pkg/machine/provider"
+	"github.com/containers/podman/v5/pkg/machine/vmconfigs"
+	"github.com/containers/podman/v5/pkg/specgen"
+	"github.com/containers/storage/pkg/fileutils"
+	"github.com/sirupsen/logrus"
+)
+
+// getMachineMounts retrieves the mounts of a machine based on the connection URI and parsed URL.
+// It returns a slice of mounts or an error if the machine cannot be found or is not running.
+func getMachineMounts(connectionURI string, parsedConnection *url.URL, machineProvider vmconfigs.VMProvider) ([]*vmconfigs.Mount, error) {
+	dirs, err := env.GetMachineDirs(machineProvider.VMType())
+	if err != nil {
+		return nil, err
+	}
+
+	machineList, err := vmconfigs.LoadMachinesInDir(dirs)
+	if err != nil {
+		return nil, fmt.Errorf("listing machines: %w", err)
+	}
+
+	// Now we know that the connection points to a machine and we
+	// can find the machine by looking for the one with the
+	// matching port.
+	connectionPort, err := strconv.Atoi(parsedConnection.Port())
+	if err != nil {
+		return nil, fmt.Errorf("parsing connection port: %w", err)
+	}
+	for _, mc := range machineList {
+		if connectionPort != mc.SSH.Port {
+			continue
+		}
+
+		state, err := machineProvider.State(mc, false)
+		if err != nil {
+			return nil, err
+		}
+
+		if state != define.Running {
+			return nil, fmt.Errorf("machine %s is not running but in state %s", mc.Name, state)
+		}
```

Comment:
This seems to duplicate code from the compose code, I suggest you split this into a function here and then make the compose code use it.

Reply:
I duplicated the code.